### ISSUE TITLE
`panic_in_result_fn` shouldn't trigger in constant environments

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#13381]
```rust
#![deny(clippy::panic_in_result_fn)]

pub fn foo<const N: usize>() {
    const {
        if N == 0 {
            panic!();
        }
    }
}
```
